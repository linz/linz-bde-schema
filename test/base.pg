\set ECHO none
--------------------------------------------------------------------------------
--
-- Copyright 2016 Crown copyright (c)
-- Land Information New Zealand and the New Zealand Government.
-- All rights reserved
--
-- This software is released under the terms of the new BSD license. See the
-- LICENSE file for more information.
--
--------------------------------------------------------------------------------
-- Provide unit testing for LINZ BDE SCHEMA using pgTAP
--------------------------------------------------------------------------------
\set QUIET true
\set VERBOSITY terse
\pset format unaligned
\pset tuples_only true

SET client_min_messages TO WARNING;

CREATE EXTENSION postgis;
CREATE EXTENSION pgtap;

--VERSIONED-- -- As of table_version 1.4.x the ver_enable_versioning function was
--VERSIONED-- -- not security definer and thus possibly lacked privileges to
--VERSIONED-- -- write in pg_depend.
--VERSIONED-- -- See https://github.com/linz/postgresql-tableversion/issues/100
--VERSIONED-- -- FIXME: drop this or make it conditionally based on
--VERSIONED-- --        table_version version when it'll be fixed
--VERSIONED-- ALTER FUNCTION table_version.ver_enable_versioning(regclass) SECURITY DEFINER;
--VERSIONED-- -- As of table_version 1.4.x the
--VERSIONED-- -- ver_versioned_table_{add,drop}_column functions were
--VERSIONED-- -- not security definers and thus possibly lacked privileges to
--VERSIONED-- -- add/drop columns in the revision tables created by
--VERSIONED-- -- security-definer ver_enable_versioning
--VERSIONED-- -- See https://github.com/linz/postgresql-tableversion/issues/113
--VERSIONED-- -- FIXME: drop this or make it conditionally based on
--VERSIONED-- --        table_version version when it'll be fixed
--VERSIONED-- ALTER FUNCTION table_version.ver_versioned_table_drop_column(name, name, name) SECURITY DEFINER;
--VERSIONED-- ALTER FUNCTION table_version.ver_versioned_table_add_column(name, name, name, text) SECURITY DEFINER;

CREATE SCHEMA _patches;
CREATE EXTENSION dbpatch SCHEMA _patches;

-- Switch to unprivileged role
DO $$
BEGIN
    IF NOT EXISTS (SELECT * FROM pg_roles
                   WHERE rolname = 'linz_bde_schema_test_role')
    THEN
        CREATE ROLE linz_bde_schema_test_role
            NOSUPERUSER CREATEDB CREATEROLE;
    END IF;

    EXECUTE format('GRANT ALL ON DATABASE %I TO linz_bde_schema_test_role',
        current_database());
    GRANT ALL ON SCHEMA _patches  TO linz_bde_schema_test_role;
    GRANT ALL ON ALL TABLES IN SCHEMA _patches  TO linz_bde_schema_test_role;

--VERSIONED-- -- NOTE: GRANT OPTION is needed so that linz_bde_schema_test_role can
--VERSIONED-- --       grant CREATE on table_version to bde_dba (as requested by
--VERSIONED-- --       sql/versioning/01-version_tables.sql)
--VERSIONED-- GRANT ALL ON SCHEMA table_version TO linz_bde_schema_test_role WITH GRANT OPTION;
    SET session_authorization TO linz_bde_schema_test_role;
END $$;



\i sql/01-bde_roles.sql
\i sql/02-bde_schema.sql
\i sql/03-bde_functions.sql
\i sql/04-bde_schema_index.sql
\i sql/05-bde_version.sql
\i sql/99-patches.sql
--VERSIONED-- \i sql/versioning/01-version_tables.sql

BEGIN;

SELECT * FROM plan(651
--VERSIONED-- + 525
);

-- Test base schema existance
\i test/base-schema.pg.inc

-- Test revision tables existance and their composition
--VERSIONED-- \i test/base-revision-tables.pg.inc

-- Test bde_write_appellation {
SELECT is(bde.bde_write_appellation
    (null,null,null,null,null,null,null,null,null,null,null,null,null,null),
    '', 'bde_write_appellation 1');
SELECT is(bde.bde_write_appellation
    (null,null,null,null,null,null,null,null,null,null,null,null,null,'14'),
    '14', 'bde_write_appellation 2');
SELECT is(bde.bde_write_appellation
    (null,null,null,null,null,null,null,null,null,null,null,null,'13','14'),
    '13 Block 14', 'bde_write_appellation 3');
SELECT is(bde.bde_write_appellation
    (null,null,null,null,null,null,null,null,null,null,null,'12','13','14'),
    '13 12 Block 14', 'bde_write_appellation 4');
SELECT is(bde.bde_write_appellation
    (null,null,null,null,null,null,null,null,null,null,'11','12','13','14'),
    '13 12 Block 14 11', 'bde_write_appellation 5');
SELECT is(bde.bde_write_appellation
    (null,null,null,null,null,null,null,null,null,'10','11','12','13','14'),
    '13 12 Block 14 11 10', 'bde_write_appellation 6');
SELECT is(bde.bde_write_appellation
    (null,null,null,null,null,null,null,null,'9','10','11','12','13','14'),
    '13 12 Block 14 11 10', 'bde_write_appellation 7');
SELECT is(bde.bde_write_appellation
    (null,null,null,null,null,null,null,'8','9','10','11','12','13','14'),
    '13 12 Block 14 Block 8 11 10', 'bde_write_appellation 8');
SELECT is(bde.bde_write_appellation
    (null,null,null,null,null,null,'7','8','9','10','11','12','13','14'),
    '13 12 Block 14 7 Block 8 11 10',
    'bde_write_appellation 9');
SELECT is(bde.bde_write_appellation
    (null,null,null,null,null,'6','7','8','9','10','11','12','13','14'),
    '13 12 Block 14 6 7 Block 8 11 10',
    'bde_write_appellation 10');
SELECT is(bde.bde_write_appellation
    (null,null,null,null,'5','6','7','8','9','10','11','12','13','14'),
    '13 12 Block 14 5 6 7 Block 8 11 10',
    'bde_write_appellation 11');
SELECT is(bde.bde_write_appellation
    (null,null,null,'4','5','6','7','8','9','10','11','12','13','14'),
    '13 12 Block 14 4 5 6 7 Block 8 11 10',
    'bde_write_appellation 12');
SELECT is(bde.bde_write_appellation
    (null,null,'3','4','5','6','7','8','9','10','11','12','13','14'),
    '3 share of 13 12 Block 14 4 5 6 7 Block 8 11 10',
    'bde_write_appellation 13');
SELECT is(bde.bde_write_appellation
    (null,'2','3','4','5','6','7','8','9','10','11','12','13','14'),
    '3 share of 13 12 Block 14 4 5 6 7 Block 8 11 10',
    'bde_write_appellation 14');
SELECT is(bde.bde_write_appellation
    ('1','2','3','4','5','6','7','8','9','10','11','12','13','14'),
    '3 share of 13 12 Block 14 4 5 6 7 Block 8 11 10',
    'bde_write_appellation 15');
SELECT is(bde.bde_write_appellation
    ('PART','2','3','4','5','6','7','8','9','10','11','12','13','14'),
    'Part3 share of 13 12 Block 14 4 5 6 7 Block 8 11 10',
    'bde_write_appellation 16');
SELECT is(bde.bde_write_appellation
    ('PART','PART','3','4','5','6','7','8','9','10','11','12','13','14'),
    'Part Part3 share of 13 12 Block 14 4 5 6 7 Block 8 11 10',
    'bde_write_appellation 17');
SELECT is(bde.bde_write_appellation
    ('PART','PART','1/1','4','5','6','7','8','9','10','11','12','13','14'),
    'Part Part 13 12 Block 14 4 5 6 7 Block 8 11 10',
    'bde_write_appellation 18');
SELECT is(bde.bde_write_appellation
    ('PART','PART','1/1','','5','6','7','8','9','10','11','12','13','14'),
    'Part Part 13 12 Block 14 5 6 7 Block 8 11 10',
    'bde_write_appellation 19');
SELECT is(bde.bde_write_appellation
    ('PART','PART','1/1','','','','7','8','9','10','11','12','13','14'),
    'Part Part 13 12 Block 14 7 Block 8 11 10',
    'bde_write_appellation 20');
SELECT is(bde.bde_write_appellation
    ('PART','PART','1/1','','','','','8','9','10','11','12','13','14'),
    'Part Part 13 12 Block 14 Block 8 11 10',
    'bde_write_appellation 21');
SELECT is(bde.bde_write_appellation
    ('PART','PART','1/1','','','','','','9','10','11','12','13','14'),
    'Part Part 13 12 Block 14 11 10',
    'bde_write_appellation 22');
SELECT is(bde.bde_write_appellation
    ('PART','PART','1/1','','','','','','PRFX','10','11','12','13','14'),
    'Part Part 13 12 Block 14 10 11',
    'bde_write_appellation 23');
SELECT is(bde.bde_write_appellation
    ('PART','PART','1/1','','','','','','PRFX','','11','12','13','14'),
    'Part Part 13 12 Block 14 11',
    'bde_write_appellation 24');
SELECT is(bde.bde_write_appellation
    ('PART','PART','1/1','','','','','','PRFX','','','12','13','14'),
    'Part Part 13 12 Block 14',
    'bde_write_appellation 25');
SELECT is(bde.bde_write_appellation
    ('PART','PART','1/1','','','','','','PRFX','','','','13','14'),
    'Part Part 13 Block 14',
    'bde_write_appellation 26');
SELECT is(bde.bde_write_appellation
    ('PART','PART','1/1','','','','','','PRFX','','','','','14'),
    'Part Part 14',
    'bde_write_appellation 27');
SELECT is(bde.bde_write_appellation
    ('PART','PART','1/1','','','','','','PRFX','','','','',''),
    'Part Part',
    'bde_write_appellation 28');
-- }

-- Test bde_get_combined_appellation {

DO $$ DECLARE tvowner TEXT; BEGIN
IF EXISTS ( SELECT * FROM pg_proc WHERE proname = 'ver_is_table_versioned' )
THEN IF table_version.ver_is_table_versioned('bde', 'crs_work') THEN

    PERFORM table_version.ver_create_revision('test fixture');

    -- As of table_version 1.4.x the triggers attached by `ver_create_revision`
    -- to tables are changed to be owned by the table owner and are made
    -- run with those permissions. This is problematic for various reasons
    -- explained in https://github.com/linz/postgresql-tableversion/issues/99
    -- As a workaround, we change those ownership right before populating one
    -- such table during testing.
    -- FIXME: drop this snippet when using a non-bogus table_version
    RESET session_authorization;
    SELECT u.rolname
      INTO tvowner
      FROM pg_roles u, pg_proc f
     WHERE f.proowner = u.oid
       AND f.proname = 'ver_create_revision'
     LIMIT 1;
    EXECUTE FORMAT('ALTER FUNCTION '
        'table_version.bde_crs_appellation_revision() OWNER TO %I',
        tvowner);
    SET session_authorization TO linz_bde_schema_test_role;

END IF; END IF; END; $$;

COPY bde.crs_appellation FROM STDIN ( delimiter '|' );
1|GNRL|Y|N|CURR|WHOL|\N|SURD|Lower Hawea|SECT|43|\N|\N|V|SUFX|\N|\N|\N|\N|\N|5084|200798
2|GNRL|Y|N|CURR|PART|\N|SURD|Clutha|SECT|3|\N|\N|XXXVI|SUFX|\N|\N|\N|\N|\N|5107|201212
3|GNRL|Y|Y|CURR|WHOL|\N|DP|13918|LOT|32|\N|\N|\N|PRFX|\N|\N|\N|\N|\N|3705368|1399023
4|GNRL|N|Y|CURR|PART|\N|DP|8007|LOT|61|\N|\N|\N|PRFX|\N|\N|\N|\N|\N|3792885|1535355
5|GNRL|Y|Y|CURR|WHOL|\N|DP|2425|LOT|11|\N|\N|VIII|PRFX|\N|\N|\N|\N|\N|3793571|1428424
6|GNRL|Y|Y|CURR|WHOL|\N|DP|28568|LOT|5|\N|\N|\N|PRFX|\N|\N|\N|\N|\N|3824354|1438685
7|GNRL|N|Y|HIST|PART|\N|DP|26040|LOT|2|\N|\N|\N|PRFX|\N|\N|\N|\N|\N|3935083|1689914
7|GNRL|Y|Y|CURR|PART|\N|DP|26042|LOT|2|\N|\N|\N|PRFX|\N|18484652|11189968|\N|\N|7187126|5831252
11|MAOR|Y|N|CURR|WHOL|Maori Reserve 883 (Rakipaoa)|\N|\N|\N|\N|\N|\N|\N|\N|\N|\N|\N|\N|\N|141034|928194
12|MAOR|Y|N|HIST|WHOL|Ohapi|\N|\N|\N|909|\N|\N|\N|\N|\N|16592162|9687268|\N|\N|6832755|5404695
13|MAOR|N|Y|CURR|WHOL|Nuhaka|\N|\N|\N|2E3D6|\N|\N|\N|\N|\N|\N|\N|\N|\N|6910776|5501059
14|MAOR|Y|Y|CURR|WHOL|Torere Pa|\N|\N|\N|12|\N|\N|\N|\N|\N|\N|\N|\N|\N|7117088|5745388
21|PART|Y|Y|CURR|WHOL|\N|DP|181158|LOT|28|\N|\N|\N|PRFX|\N|\N|\N|\N|\N|6740001|5278098
31|OTHR|N|Y|CURR|WHOL|\N|\N|\N|\N|\N|\N|\N|\N|\N|Stopped Road|\N|\N|\N|\N|3000867|289
32|OTHR|N|Y|HIST|PART|\N|\N|\N|\N|\N|\N|\N|\N|\N|Closed Street Block V Lower Kaikorai Survey District|\N|\N|\N|\N|3001281|427
33|OTHR|N|Y|CURR|PART|\N|\N|\N|\N|\N|\N|\N|\N|\N|Section 44 River Sections East Taieri Survey District|\N|\N|\N|\N|3001833|611
34|OTHR|Y|Y|CURR|PART|\N|\N|\N|\N|\N|\N|\N|\N|\N|Closed Road Block II Clutha Survey District|\N|\N|\N|\N|3041661|13887
35|OTHR|Y|Y|CURR|WHOL|\N|\N|\N|\N|\N|\N|\N|\N|\N|Section 102 Irregular Block East Taieri Survey District|\N|\N|\N|\N|3044865|14955
36|OTHR|Y|N|CURR|WHOL|\N|\N|\N|\N|\N|\N|\N|\N|\N|Future Development Accessory Unit 9A Deposited Plan 22767|\N|\N|\N|\N|443|173809
\.

DO $$ BEGIN
IF EXISTS ( SELECT * FROM pg_proc WHERE proname = 'ver_is_table_versioned' )
THEN IF table_version.ver_is_table_versioned('bde', 'crs_work') THEN
    PERFORM table_version.ver_complete_revision();
END IF; END IF; END; $$;

SELECT results_eq($$
	SELECT DISTINCT par_id,
		   bde.bde_get_combined_appellation(par_id, 'N') app_short,
		   bde.bde_get_combined_appellation(par_id, 'Y') app_long
	FROM bde.crs_appellation
	ORDER BY par_id
$$, $$ VALUES
	(1,'43 Block V Lower Hawea','43 Block V Lower Hawea'),
	(2,'Part 3 Block XXXVI Clutha','Part 3 Block XXXVI Clutha'),
	(3,'32 DP 13918','32 13918'),
	(4,'Part 61 DP 8007','Part 61 8007'),
	(5,'11 Block VIII DP 2425','11 Block VIII 2425'),
	(6,'5 DP 28568','5 28568'),
	(7,'Part 2 DP 26042','Part 2 26042'),
	(11,'Maori Reserve 883 (Rakipaoa) Block','Maori Reserve 883 (Rakipaoa) Block'),
	(12,null,null),
	(13,'Nuhaka 2E3D6 Block','Nuhaka 2E3D6 Block'),
	(14,'Torere Pa 12 Block','Torere Pa 12 Block'),
	(21,'28 Block 28 DP 181158','28 Block 28 181158'),
	(31,'Stopped Road','Stopped Road'),
	(32,null,null),
	(33,'Part Section 44 River Sections East Taieri Survey District','Part Section 44 River Sections East Taieri Survey District'),
	(34,'Part Closed Road Block II Clutha Survey District','Part Closed Road Block II Clutha Survey District'),
	(35,'Section 102 Irregular Block East Taieri Survey District','Section 102 Irregular Block East Taieri Survey District'),
	(36,'Future Development Accessory Unit 9A Deposited Plan 22767','Future Development Accessory Unit 9A Deposited Plan 22767')
$$,
	'bde_get_combined_appellation should behave as expected'
);

-- }

SELECT * FROM finish();

ROLLBACK;

-- Cleanup
DO $$
BEGIN
    RESET session_authorization;
    EXECUTE format('REASSIGN OWNED BY linz_bde_schema_test_role TO %I',
        current_user);
    EXECUTE format('REVOKE ALL ON DATABASE %I FROM linz_bde_schema_test_role',
        current_database());
    REVOKE ALL ON SCHEMA _patches  FROM linz_bde_schema_test_role;
    REVOKE ALL ON ALL TABLES IN SCHEMA _patches FROM linz_bde_schema_test_role;
--VERSIONED-- -- TODO: figure out which "dependent privileges exist", to
--VERSIONED-- --      replace the CASCADE with fine-grained REVOKEs
--VERSIONED-- REVOKE ALL ON SCHEMA table_version FROM linz_bde_schema_test_role CASCADE;
    DROP ROLE linz_bde_schema_test_role;
END; $$;


