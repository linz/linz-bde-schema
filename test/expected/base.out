\set ECHO none
psql:sql/99-patches.sql:60: INFO:  Applying patch BDE - 1.0.2: Remove annotations column from bde.crs_work
ok 1 - Schema bde should exist
ok 2 - Role bde_dba should exist
ok 3 - Role bde_admin should exist
ok 4 - Role bde_user should exist
ok 5 - Table bde.cbe_title_parcel_association should exist
ok 6 - Table bde.crs_action should exist
ok 7 - Table bde.crs_action_type should exist
ok 8 - Table bde.crs_adj_obs_change should exist
ok 9 - Table bde.crs_adj_user_coef should exist
ok 10 - Table bde.crs_adjust_coef should exist
ok 11 - Table bde.crs_adjust_method should exist
ok 12 - Table bde.crs_adjustment_run should exist
ok 13 - Table bde.crs_adoption should exist
ok 14 - Table bde.crs_affected_parcl should exist
ok 15 - Table bde.crs_alias should exist
ok 16 - Table bde.crs_appellation should exist
ok 17 - Table bde.crs_comprised_in should exist
ok 18 - Table bde.crs_coordinate should exist
ok 19 - Table bde.crs_coordinate_sys should exist
ok 20 - Table bde.crs_coordinate_tpe should exist
ok 21 - Table bde.crs_cor_precision should exist
ok 22 - Table bde.crs_cord_order should exist
ok 23 - Table bde.crs_datum should exist
ok 24 - Table bde.crs_elect_place should exist
ok 25 - Table bde.crs_ellipsoid should exist
ok 26 - Table bde.crs_enc_share should exist
ok 27 - Table bde.crs_encumbrance should exist
ok 28 - Table bde.crs_encumbrancee should exist
ok 29 - Table bde.crs_estate_share should exist
ok 30 - Table bde.crs_feature_name should exist
ok 31 - Table bde.crs_geodetic_network should exist
ok 32 - Table bde.crs_geodetic_node_network should exist
ok 33 - Table bde.crs_image should exist
ok 34 - Table bde.crs_image_history should exist
ok 35 - Table bde.crs_land_district should exist
ok 36 - Table bde.crs_legal_desc should exist
ok 37 - Table bde.crs_legal_desc_prl should exist
ok 38 - Table bde.crs_line should exist
ok 39 - Table bde.crs_locality should exist
ok 40 - Table bde.crs_maintenance should exist
ok 41 - Table bde.crs_map_grid should exist
ok 42 - Table bde.crs_mark should exist
ok 43 - Table bde.crs_mark_name should exist
ok 44 - Table bde.crs_mark_sup_doc should exist
ok 45 - Table bde.crs_mesh_blk should exist
ok 46 - Table bde.crs_mesh_blk_area should exist
ok 47 - Table bde.crs_mesh_blk_bdry should exist
ok 48 - Table bde.crs_mesh_blk_line should exist
ok 49 - Table bde.crs_mesh_blk_place should exist
ok 50 - Table bde.crs_mrk_phys_state should exist
ok 51 - Table bde.crs_network_plan should exist
ok 52 - Table bde.crs_node should exist
ok 53 - Table bde.crs_node_prp_order should exist
ok 54 - Table bde.crs_node_works should exist
ok 55 - Table bde.crs_nominal_index should exist
ok 56 - Table bde.crs_obs_accuracy should exist
ok 57 - Table bde.crs_obs_elem_type should exist
ok 58 - Table bde.crs_obs_set should exist
ok 59 - Table bde.crs_obs_type should exist
ok 60 - Table bde.crs_observation should exist
ok 61 - Table bde.crs_off_cord_sys should exist
ok 62 - Table bde.crs_office should exist
ok 63 - Table bde.crs_ordinate_adj should exist
ok 64 - Table bde.crs_ordinate_type should exist
ok 65 - Table bde.crs_parcel should exist
ok 66 - Table bde.crs_parcel_bndry should exist
ok 67 - Table bde.crs_parcel_dimen should exist
ok 68 - Table bde.crs_parcel_label should exist
ok 69 - Table bde.crs_parcel_ring should exist
ok 70 - Table bde.crs_programme should exist
ok 71 - Table bde.crs_proprietor should exist
ok 72 - Table bde.crs_reduct_meth should exist
ok 73 - Table bde.crs_reduct_run should exist
ok 74 - Table bde.crs_ref_survey should exist
ok 75 - Table bde.crs_road_ctr_line should exist
ok 76 - Table bde.crs_road_name should exist
ok 77 - Table bde.crs_road_name_asc should exist
ok 78 - Table bde.crs_setup should exist
ok 79 - Table bde.crs_site should exist
ok 80 - Table bde.crs_site_locality should exist
ok 81 - Table bde.crs_stat_act_parcl should exist
ok 82 - Table bde.crs_stat_version should exist
ok 83 - Table bde.crs_statist_area should exist
ok 84 - Table bde.crs_statute should exist
ok 85 - Table bde.crs_statute_action should exist
ok 86 - Table bde.crs_street_address should exist
ok 87 - Table bde.crs_sur_admin_area should exist
ok 88 - Table bde.crs_sur_plan_ref should exist
ok 89 - Table bde.crs_survey should exist
ok 90 - Table bde.crs_survey_image should exist
ok 91 - Table bde.crs_sys_code should exist
ok 92 - Table bde.crs_sys_code_group should exist
ok 93 - Table bde.crs_title should exist
ok 94 - Table bde.crs_title_action should exist
ok 95 - Table bde.crs_title_doc_ref should exist
ok 96 - Table bde.crs_title_estate should exist
ok 97 - Table bde.crs_title_mem_text should exist
ok 98 - Table bde.crs_title_memorial should exist
ok 99 - Table bde.crs_topology_class should exist
ok 100 - Table bde.crs_transact_type should exist
ok 101 - Table bde.crs_ttl_enc should exist
ok 102 - Table bde.crs_ttl_hierarchy should exist
ok 103 - Table bde.crs_ttl_inst should exist
ok 104 - Table bde.crs_ttl_inst_title should exist
ok 105 - Table bde.crs_unit_of_meas should exist
ok 106 - Table bde.crs_user should exist
ok 107 - Table bde.crs_vector should exist
ok 108 - Table bde.crs_vertx_sequence should exist
ok 109 - Table bde.crs_work should exist
ok 110 - Function bde.bde_get_app_specific() should exist
ok 111 - Function bde.bde_get_charcode() should exist
ok 112 - Function bde.bde_get_dataset() should exist
ok 113 - Function bde.bde_get_datasource() should exist
ok 114 - Function bde.bde_get_desccode() should exist
ok 115 - Function bde.bde_get_district() should exist
ok 116 - Function bde.bde_get_mrknodename() should exist
ok 117 - Function bde.bde_get_markname() should exist
ok 118 - Function bde.bde_get_nodename() should exist
ok 119 - Function bde.bde_get_nodename() should exist
ok 120 - Function bde.bde_get_nodeorder() should exist
ok 121 - Function bde.bde_get_office() should exist
ok 122 - Function bde.bde_get_roadname() should exist
ok 123 - Function bde.bde_get_trtcode() should exist
ok 124 - Function bde.bde_get_username() should exist
ok 125 - Function bde.bde_get_useroff() should exist
ok 126 - Function bde.bde_write_appellation() should exist
ok 127 - Function bde.bde_get_combined_appellation() should exist
ok 128 - Function bde.bde_get_par_stat_act() should exist
ok 129 - Function bde.bde_getnearnodes() should exist
ok 130 - Function bde.bde_drop_idle_connections() should exist
ok 131 - Function bde.bde_version() should exist
ok 132 - Index fk_app_par should exist
ok 133 - Index shx_epl_shape should exist
ok 134 - Index shx_fen_shape should exist
ok 135 - Index shx_ldt_shape should exist
ok 136 - Index shx_lin_shape should exist
ok 137 - Index shx_loc_shape should exist
ok 138 - Index shx_map_shape should exist
ok 139 - Index fk_mkn_mrk should exist
ok 140 - Index shx_mbk_shape should exist
ok 141 - Index shx_mbl_shape should exist
ok 142 - Index shx_nod_shape should exist
ok 143 - Index shx_ocs_shape should exist
ok 144 - Index shx_par_shape should exist
ok 145 - Index shx_plb_shape should exist
ok 146 - Index shx_rcl_shape should exist
ok 147 - Index fk_sap_sta should exist
ok 148 - Index fk_sap_par should exist
ok 149 - Index shx_stt_shape should exist
ok 150 - Index shx_sad_shape should exist
ok 151 - Index shx_spf_shape should exist
ok 152 - Index fk_sco_scg should exist
ok 153 - Index fk_sco_scg_code should exist
ok 154 - Index shx_vct_shape should exist
1..154
